### DISCUSSION TITLE
Collecting page data .SyntaxError: Unexpected token u in JSON at position 0

### DISCUSSION THREAD

**Discussion by coding-to-music** [vercel/vercel#9006] [Help] [answered]
Hello, my repo is here: https://github.com/coding-to-music/swejobs-next-firebase  

I am attempting to build a project I found on GitHub and there is an error trying to collect page data - so not sure if it is something in my setup or if the people who wrote the original repo ran into similar issues. 

I am unable to run a build locally and also Vercel has the error too. 

## Error creating a build

```
npm run build
```

Output

```
> swejobs-next-firebase@0.1.0 build
> next build

info  - Loaded env from /mnt/volume_nyc1_01/swejobs-next-firebase/.env
info  - Linting and checking validity of types
info  - Creating an optimized production build
info  - Compiled successfully
info  - Collecting page data .SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
    at /mnt/volume_nyc1_01/swejobs-next-firebase/.next/server/chunks/186.js:961:27

> Build error occurred
Error: Failed to collect page data for /about/about-data
    at /mnt/volume_nyc1_01/swejobs-next-firebase/node_modules/next/dist/build/utils.js:959:15
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  type: 'Error'
}
info  - Collecting page data .
```

## Error when Vercel tries to do a build

```
info  - Linting and checking validity of types...
info  - Creating an optimized production build...
info  - Compiled successfully
info  - Collecting page data...
SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
    at /vercel/path0/.next/server/chunks/186.js:961:27
> Build error occurred
Error: Failed to collect page data for /about/about-data
    at /vercel/path0/node_modules/next/dist/build/utils.js:959:15 {
  type: 'Error'
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Command "yarn run build" exited with 1
```

## Versions

```
node --version
v16.14.2

npm --version
9.1.1

vercel --version
Vercel CLI 28.5.6
```

**Comment by amyegan** [comment, marked as answer]
Thanks for sharing the repo @coding-to-music! Have you made sure that all of the [expected environment variables](https://github.com/coding-to-music/swejobs-next-firebase/blob/main/.env-example) are set in your project settings? This might be related to missing env vars, or it may be a code problem. 

If updating the environment variabless doesn't help, you should also [ask the Next.js community](https://github.com/vercel/next.js/discussions) since it could be a code error and they might be able to offer a quick solution.

**Reply by coding-to-music** [reply]
Thanks I have now asked over in the Next.js community https://github.com/vercel/next.js/discussions/43833
